Library/Homebrew/cleanup.rb

```diff
@@ -36,10 +42,18 @@ def self.cleanup_logs
       end
     end
 
-    def self.cleanup_cellar
-      Formula.installed.each do |formula|
-        cleanup_formula formula
+    def self.cleanup_cellar(formulae = Formula.installed)
+      unremovable_kegs = []
+      formulae.each do |formula|
+        cleanup_formula formula, unremovable_kegs
```

Comment:
Using an instance variable also allows us to move the failure messaging to the command rather than the library functions. We can also continue onto the cache, logs, lockfiles etc when formula cleanup fails.